pkg/controllers/leasegarbagecollection/controller.go

```diff
@@ -0,0 +1,67 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package leasegarbagecollection
+
+import (
+	"context"
+
+	v1 "k8s.io/api/coordination/v1"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	corecontroller "github.com/aws/karpenter-core/pkg/operator/controller"
+)
+
+// Controller for the resource
+type Controller struct {
+	kubeClient client.Client
+}
+
+// NewController is a constructor
+func NewController(kubeClient client.Client) corecontroller.Controller {
+	return corecontroller.Typed[*v1.Lease](kubeClient, &Controller{
+		kubeClient: kubeClient,
+	})
+}
+
+func (c *Controller) Name() string {
+	return "lease.garbagecollection"
+}
+
+// Reconcile the resource
+func (c *Controller) Reconcile(ctx context.Context, l *v1.Lease) (reconcile.Result, error) {
+	if l.OwnerReferences != nil || l.Namespace != "kube-node-lease" {
```

Comment:
Are there any other traits of the lease that ensure that we aren't messing with other leases that have no owner references in this namespace?

Reply:
We could also limit it also nodes karpenter has launched

Reply:
How are we able to detect if it's from a node that Karpenter launched if it has no owner references? I don't think that's possible.

Reply:
I was thinking we could get the instance from EC2 and look at the tags. Since the name of the lease is the same as the node

Reply:
Maybe if there's a creationTimestamp we can ensure that it's at least an hour old so we don't run into race conditions?

Reply:
It also look like the ownerReference is add on the lease creation

https://github.com/kubernetes/kubernetes/blob/b1e35d5616e066c4b0b3f550c33bbf67bb3f876e/staging/src/k8s.io/component-helpers/apimachinery/lease/controller.go#L210